base/shared/src/main/scala/scalaz/tc/band.scala

```diff
@@ -0,0 +1,14 @@
+package scalaz.tc
+
+import scala.Int
+
+/**
+ * Band is a semigroup with idempotency.
+ *
+ */
+trait BandClass[A] extends SemigroupClass[A] {
+  override def exponent(a: A, i: Int): A = {
+    val _ = i // allocates, unfortunately - if we can remove this, that would be great
```

Comment:
This doesn't allocate anything on the heap, just on the stack. Really not a problem.